File: client/scala-gatling/src/gatling/scala/org/openapitools/client/model/Query.scala

package org.openapitools.client.model


case class Query (
    /* Unique study key */
    _studyKey: Option[String],
    /* Mednet subject ID associated with the query (if applicable) */
    _subjectId: Option[Integer],
    /* Client-assigned subject OID (if applicable) */
    _subjectOid: Option[String],
    /* Category/type of the query (e.g., subject, record, question) */
    _annotationType: Option[String],
    /* Unique system identifier for the query */
    _annotationId: Option[Integer],
    /* System text identifier for the query type/location (subject, record, question) */
    _type: Option[String],
    /* Description of the query (e.g., reason or context) */
    _description: Option[String],
    /* Record ID associated with the query (if applicable) */
    _recordId: Option[Integer],
    /* Variable name (field) associated with the query (if applicable) */
    _variable: Option[String],
    /* Subject display ID associated with the query (if applicable) */
    _subjectKey: Option[String],
    /* Date when the query was created */
    _dateCreated: Option[String],
    /* Date when the query was last modified */
    _dateModified: Option[String],
    /* History of comments/actions on the query */
    _queryComments: Option[List[QueryComment]]
)
object Query {
    def toStringBody(var_studyKey: Object, var_subjectId: Object, var_subjectOid: Object, var_annotationType: Object, var_annotationId: Object, var_type: Object, var_description: Object, var_recordId: Object, var_variable: Object, var_subjectKey: Object, var_dateCreated: Object, var_dateModified: Object, var_queryComments: Object) =
        s"""
        | {
        | "studyKey":$var_studyKey,"subjectId":$var_subjectId,"subjectOid":$var_subjectOid,"annotationType":$var_annotationType,"annotationId":$var_annotationId,"type":$var_type,"description":$var_description,"recordId":$var_recordId,"variable":$var_variable,"subjectKey":$var_subjectKey,"dateCreated":$var_dateCreated,"dateModified":$var_dateModified,"queryComments":$var_queryComments
        | }
        """.stripMargin
}

File: models/Query.js
const utils = require('../utils/utils');
const QueryComment = require('../models/QueryComment');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}studyKey`,
                label: `Unique study key - [${labelPrefix}studyKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}subjectId`,
                label: `Mednet subject ID associated with the query (if applicable) - [${labelPrefix}subjectId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}subjectOid`,
                label: `Client-assigned subject OID (if applicable) - [${labelPrefix}subjectOid]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}annotationType`,
                label: `Category/type of the query (e.g., subject, record, question) - [${labelPrefix}annotationType]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}annotationId`,
                label: `Unique system identifier for the query - [${labelPrefix}annotationId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}type`,
                label: `System text identifier for the query type/location (subject, record, question) - [${labelPrefix}type]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}description`,
                label: `Description of the query (e.g., reason or context) - [${labelPrefix}description]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}recordId`,
                label: `Record ID associated with the query (if applicable) - [${labelPrefix}recordId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}variable`,
                label: `Variable name (field) associated with the query (if applicable) - [${labelPrefix}variable]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}subjectKey`,
                label: `Subject display ID associated with the query (if applicable) - [${labelPrefix}subjectKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateCreated`,
                label: `Date when the query was created - [${labelPrefix}dateCreated]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateModified`,
                label: `Date when the query was last modified - [${labelPrefix}dateModified]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}queryComments`,
                label: `[${labelPrefix}queryComments]`,
                children: QueryComment.fields(`${keyPrefix}queryComments${!isInput ? '[]' : ''}`, isInput, true), 
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'studyKey': bundle.inputData?.[`${keyPrefix}studyKey`],
            'subjectId': bundle.inputData?.[`${keyPrefix}subjectId`],
            'subjectOid': bundle.inputData?.[`${keyPrefix}subjectOid`],
            'annotationType': bundle.inputData?.[`${keyPrefix}annotationType`],
            'annotationId': bundle.inputData?.[`${keyPrefix}annotationId`],
            'type': bundle.inputData?.[`${keyPrefix}type`],
            'description': bundle.inputData?.[`${keyPrefix}description`],
            'recordId': bundle.inputData?.[`${keyPrefix}recordId`],
            'variable': bundle.inputData?.[`${keyPrefix}variable`],
            'subjectKey': bundle.inputData?.[`${keyPrefix}subjectKey`],
            'dateCreated': bundle.inputData?.[`${keyPrefix}dateCreated`],
            'dateModified': bundle.inputData?.[`${keyPrefix}dateModified`],
            'queryComments': utils.childMapping(bundle.inputData?.[`${keyPrefix}queryComments`], `${keyPrefix}queryComments`, QueryComment),
        }
    },
}
